dbms/src/Storages/DeltaMerge/DeltaMergeHelpers.cpp

```diff
@@ -28,8 +28,8 @@ Digest hashSchema(const Block & schema)
     const auto & data = schema.getColumnsWithTypeAndName();
     for (const auto & column_with_type_and_name : data)
     {
-        const auto & type = column_with_type_and_name.type->getTypeId();
-        SHA256_Update(&ctx, reinterpret_cast<const unsigned char *>(&type), sizeof(type));
+        const auto & type = column_with_type_and_name.type->getName();
+        SHA256_Update(&ctx, reinterpret_cast<const unsigned char *>(type.c_str()), type.size());
```

Comment:
I didn't get it, why will it cause hash problems?

Reply:
When the datatype of Table A's column a is Nullable(decimal(20, 6)), and the datatype of Table B's column a is Nullable(datetime(6))

Reply:
Better add comments in the codes to explain the reason

Reply:
Get, I add some comments for it.